podbuilder/podbuilder.go

```diff
@@ -152,6 +153,23 @@ func (pb *PodBuilder) PodForClient(client *grpcv1.Client) (*corev1.Pod, error) {
 		Name:  config.DriverPortEnv,
 		Value: fmt.Sprint(config.DriverPort)})
 
+	if xdsServer := kubehelpers.ContainerForName("xds-server", pod.Spec.Containers); xdsServer != nil {
+		if envoy := kubehelpers.ContainerForName("envoy", pod.Spec.Containers); envoy == nil {
+			pod.Spec.Volumes = append(pod.Spec.Volumes, corev1.Volume{Name: "grpc-xds-bootstrap"})
+
+			runContainer.VolumeMounts = append(runContainer.VolumeMounts, corev1.VolumeMount{
+				Name:      "grpc-xds-bootstrap",
+				MountPath: "/bootstrap",
+				ReadOnly:  false,
+			})
+			xdsServer.VolumeMounts = append(xdsServer.VolumeMounts, corev1.VolumeMount{
+				Name:      "grpc-xds-bootstrap",
+				MountPath: "/bootstrap",
```

Comment:
This is unexpected to me. How is the xDS server using the bootstrap file?

Reply:
The clients need the bootstrap, however, it would not be very convenient to rebuilt each worker's image to include bootstrap file, better options is to have the bootstrap file in a more centralized image, either the clone image (worker's init container) or xDS server image. 
We used xDS server's image instead of the clone image to minimized the change for a regular test (i.e. for people who doesn't run PSM test, if they see the logic of potentially moving a file in the clone image, they might be confused). Also we always need to have xDS server when running PSM tests, the xDS server image makes a better choice to store the bootstrap file.

Since the bootstrap file is stored inside the xDS image, I need to move the bootstrap file to a shared volume (between xds server and the client) so that the client would have access to it. Let me know if I have not explained it well. :)

Reply:
I see. That makes sense. But both of these mounts have `ReadOnly: false`? Based on your explanation above, the `runContainer` only _reads_ the bootstrap file -- it doesn't write to it. So shouldn't `ReadOnly: true` be set?

Reply:
I can make them read only: true.

Reply:
Right, I think it is okay to have readonly for run container. Thanks!